src/30_Advanced/favorite.html

```diff
@@ -0,0 +1,117 @@
+  <!--
+  Customize success or error messages by using the [`.amp-form-submit-success` and `.amp-form-submit-error` CSS
+  classes](https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#classes-and-css-hooks).
+  For success, we'll change the color of the heart icon, using CSS filters
+  ([credit](http://codepen.io/noahblon/post/coloring-svgs-in-css-background-images)). Fiters are supported by most
+  browsers except IE and Edge, for which the SVG sprite method could be used instead, or even the ♥ Unicode character.
+
+  The example works by having a form submit button without text, and styled with a background image pointing
+  to an SVG with the icon we want (a heart in this example). Once the form is submitted, we'll hide the button and
+  display a `<div submit-success>`, which we style with the same background image, but a different color.
+  -->
+  <style amp-custom>
+    form.amp-form-submit-success [submit-success] {
+      /* Change the heart color from the SVG red to orange */
+      filter: hue-rotate(40deg) saturate(0.5) brightness(390%) saturate(4);
+      /* The Android browser requires a prefix: http://caniuse.com/#search=css%20filter */
+      -webkit-filter: hue-rotate(40deg) saturate(0.5) brightness(390%) saturate(4);
+    }
+    form.amp-form-submit-error [submit-error] {
+      color: red;
+    }
+
+    .favorite {
+      width: 48px;
+      height: 48px;
+      display: inline-block;
+      cursor: pointer;
+
+      background: url(https://upload.wikimedia.org/wikipedia/commons/f/f1/Heart_coraz%C3%B3n.svg) no-repeat 50% 50%;
+      background-size: cover;
+
+      /* Turn the red SVG into grey */
+      filter: grayscale(100%) brightness(5) brightness(.9);
+      -webkit-filter: grayscale(100%) brightness(5) brightness(.9);
+
+      /* Disable any border or outline the form > input element might have */
+      outline: none;
+      border: none;
+    }
+  </style>
+</head>
+
+<body>
+  <!-- #### Submitting the form via XMLHttpRequest
+
+  Since we've loaded `<amp-form>`, we can use a regular `<form>` element. It only has one `input` control,
+  the submit button, and an `action-xhr` attribute set to the server endpoint. This prevents the form submission from
+  loading a new page. The endpoint result will be egnored for the purposes of this example, as well as [the
+  target attribute](https://github.com/ampproject/amphtml/issues/5607#issuecomment-253894602).
+
+  #### Replacing the "favorite" input field with a "favorite<b>d</b>" icon
+
+  The form's [on
+  attribute](https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#events) is set to
+  call the `hide` built-in method of the `input` (identified by its id, `favorite-btn`). If the submission succeeds,
+
+  1. The [`<div submit-success>` element will be
+     rendered](https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#successerror-response-rendering).
```

Comment:
we are trying to link to the docs as much as possible instead of linking to the github project. In here [link](https://www.ampproject.org/docs/reference/components/amp-form) can be used, I don't think it supports the anchor though